prog/prog.go

```diff
@@ -79,6 +79,46 @@ func (a *Arg) InnerArg(typ sys.Type) *Arg {
 	}
 }
 
+func encodeValue(value uintptr, size uintptr, end sys.Endianness) uintptr {
+	if end == sys.LittleEndian {
+		return value
+	}
+	if size == 0 {
+	}
+	switch size {
+	case 0:
+		// This happens when a syscall has a big endian argument.
```

Comment:
Let's give proper size to all types, including syscall arguments.
I don't see why int32be should be prohibited in syscall aguments, but allowed everywhere else. Moreover, syscall arguments do have size -- they are intptr's effectively. Sysgen already makes assumptions about intptr size in `sz := int64(64) // TODO: assume that pointer is 8 bytes for now`. Add `const ptrSize = 8` to sysgen.go and use to give size to all types where size is 0 currently.